docs/website/docs/general-usage/data-enrichments/url-parser-data-enrichment.md

```diff
@@ -0,0 +1,247 @@
+
+       This function sends a URL to a specified API endpoint for URL parsing.
+
+       Parameters:
+       url (str): The URL to be parsed.
+
+       Returns:
+       dict: Parsed URL data in JSON format if the request is successful.
+       None: If the request fails (e.g., an invalid URL or server error).
+       """
+       # Define the API endpoint URL for the URL parsing service
+       api_url = "https://api.urlparse.com/v1/query"
+       url = record['page_referer']
+       # Send a POST request to the API with the URL to be parsed
+       response = requests.post(api_url, json={"url": url})
+
+       # Check if the response from the API is successful (HTTP status code 200)
+       if response.status_code == 200:
+           # If successful, return the parsed data in JSON format
+           return response.json()
+       else:
+           # If the request failed, print an error message with the status code and return None
+           print(f"Request for {url} failed with status code: {response.status_code}")
+           return None
+   ```
+
+### 3. Create your pipeline
+
+1. In creating the pipeline, the `url_parser` can be used in the following ways:
+
+   - Add map function
+   - Transformer function
+
+   The `dlt` library's `transformer` and `add_map` functions serve distinct purposes in data
+   processing.
+
+   `Transformers` used to process a resource and are ideal for post-load data transformations in a
+   pipeline, compatible with tools like `dbt`, the `dlt SQL client`, or Pandas for intricate data
+   manipulation. To read more:
+   [Click here.](../../general-usage/resource.md#process-resources-with-dlttransformer)
+
+   Conversely, `add_map` used to customize a resource applies transformations at an item level
+   within a resource. It's useful for tasks like anonymizing individual data records. More on this
+   can be found under [Customize resources](../../general-usage/resource.md#customize-resources) in
+   the documentation.
+
+1. Here, we create the pipeline and use the `add_map` functionality:
+
+   ```python
+   # Create the pipeline
+   pipeline = dlt.pipeline(
+       pipeline_name="data_enrichment_three",
+       destination="duckdb",
+       dataset_name="user_device_enrichment",
+   )
+
+   # Run the pipeline with the transformed source
+   load_info = pipeline.run(tracked_data.add_map(url_parser))
+
+   print(load_info)
+   ```
+
+   :::info Please note that the same outcome can be achieved by using the transformer function. To
+   do so, you need to add the transformer decorator at the top of the `url_parser` function. For
+   `pipeline.run`, you can use the following code:
+
+   ```python
+   # using fetch_average_price as a transformer function
+   load_info = pipeline.run(
+       tracked_data | url_parser,
+       table_name="url_parser"
+   )
+   ```
+
+   This will execute the `url_parser` function with the tracked data and return parsed URL. :::
```

Comment:
move ::: to the next line
![image](https://github.com/dlt-hub/dlt/assets/20367975/ca340f55-b9ea-4163-932a-1f6d770a082b)